From 089fa4e1c33187e4bff15eb7d7659e1556a82e82 Mon Sep 17 00:00:00 2001
From: Raveendra Talabattula <raveendra.talabattula@amd.com>
Date: Wed, 28 Mar 2018 17:10:39 +0530
Subject: [PATCH 3316/5855] drm/syncobj: increase the fence reference count

Increase the reference count of the fence associated with syncobj
inorder to access it.This fixes the hang issue observed with
Vulkan Hologram sample run.

Signed-off-by: Raveendra Talabattula <raveendra.talabattula@amd.com>
---
 drivers/gpu/drm/drm_syncobj.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/drm_syncobj.c b/drivers/gpu/drm/drm_syncobj.c
index ecc3682..3a24884 100644
--- a/drivers/gpu/drm/drm_syncobj.c
+++ b/drivers/gpu/drm/drm_syncobj.c
@@ -238,6 +238,10 @@ int drm_syncobj_find_fence(struct drm_file *file_private,
 		return -ENOENT;
 
 	*fence = drm_syncobj_fence_get(syncobj);
+	
+	if (syncobj->fence){
+		*fence = fence_get(syncobj->fence);
+	}
 	if (!*fence) {
 		ret = -EINVAL;
 	}
-- 
2.7.4

